src/browser-driver-manager.js

```diff
@@ -1,259 +1,175 @@
- * @param {Boolean} [options.sudo] - Run the command as sudo
- * @param {String[]} options.args - List of script arguments
- * @returns {Promise<String|null>}
- */
-function runBashScript({ returnValue = false, sudo = false, args } = {}) {
-  return new Promise((resolve, reject) => {
-    const lines = [];
-    let lastChunk;
+async function install(browserId, options) {
+  // When parsing the values the version value could be set
+  // as a version number (e.g. 116.0.5845.96) or a channel (e.g. beta, dev, canary)
+  const [browser, version = 'latest'] = browserId.split('@');
 
-    const scriptPath = path.join(__dirname, 'index.sh');
-    let child;
-    if (sudo) {
-      child = spawn('sudo', [scriptPath, ...args]);
-    } else {
-      child = spawn(scriptPath, args);
-    }
+  // Create a cache directory if it does not exist on the user's home directory
+  // This will be where environment variables will be stored for the tests
+  // since it is a consistent location across different platforms
+  const BDMCacheDir = getBDMCacheDir();
+  try {
+    await fsPromises.mkdir(BDMCacheDir, { recursive: true });
+  } catch {
+    console.log(`${BDMCacheDir} already exists`);
+  }
 
-    child.stdout.on('data', chunk => {
-      lines.push(chunk);
+  // Should support for other browsers be added, commander should handle this check.
+  // With only one supported browser, this error message is more meaningful than commander's.
+  if (!browser.includes('chrome')) {
+    throw new Error(
+      `The selected browser, ${browser}, could not be installed. Currently, only "chrome" is supported.`
+    );
+  }
 
-      if (!returnValue) {
-        process.stdout.write(chunk);
-      }
-    });
+  const platform = detectBrowserPlatform();
 
-    // for some reason the install script will output curl download
-    // percent through stderr
-    child.stderr.on('data', chunk => {
-      // mimic download percent bar by manually clearing lines
-      // @see https://stackoverflow.com/questions/17309749/node-js-console-log-is-it-possible-to-update-a-line-rather-than-create-a-new-l
-      if (chunk.indexOf('\r') !== -1) {
-        process.stdout.clearLine();
-      }
-      if (lastChunk) {
-        process.stdout.cursorTo(lastChunk.length - 1);
-      }
-      process.stdout.write(chunk);
+  if (!platform) {
+    throw new Error('Unable to detect browser platform');
+  }
+  // This will sync the browser and chromedriver versions
+  let chromeBuildId;
+  try {
+    chromeBuildId = await resolveBuildId(Browser.CHROME, platform, version);
+  } catch (e) {
+    throw new Error(e);
+  }
 
-      lastChunk = chunk;
+  const currentVersion = await getVersion();
+  if (currentVersion) {
+    if (currentVersion === chromeBuildId) {
+      console.log(
+        `Chrome and Chromedriver versions ${currentVersion} are already installed. Skipping installation.`
+      );
+      return;
+    }
+    console.log(
```

Comment:
In this case, should we remove previously installed versions?

Reply:
I think I like the idea of cleaning up unused versions. Otherwise after 5 installs of different versions you have 5 chromes and chromedrivers in your directory, resulting in 1.1G of data (each chrome is ~250M)